src/sentry/static/sentry/app/components/gridEditable/index.tsx

```diff
@@ -92,47 +86,25 @@ type GridEditableProps<DataRow, ColumnKey> = {
     prependColumnWidths?: string[];
   };
 
-  /**
-   * As GridEditable is unopinionated about the structure of GridColumn,
-   * ModalEditColumn relies on the parent component to provide the form layout
-   * and logic to create/update the columns
-   */
-  modalEditColumn: {
-    renderBodyWithForm: (
-      indexColumnOrder?: number,
-      column?: GridColumn<ColumnKey>,
-      onSubmit?: (column: GridColumn<ColumnKey>) => void,
-      onSuccess?: () => void,
-      onError?: () => void
-    ) => React.ReactNode;
-    renderFooter: () => React.ReactNode;
-  };
-
   /**
    * As there is no internal state being maintained, the parent component will
    * have to provide functions to update the state of the columns, especially
    * after moving/resizing
    */
   actions: {
-    moveColumnCommit: (indexFrom: number, indexTo: number) => void;
-    onDragStart: (
-      event: React.MouseEvent<SVGSVGElement, MouseEvent>,
-      indexFrom: number
-    ) => void;
-    deleteColumn: (index: number) => void;
+    editColumns: () => void;
     downloadAsCsv: () => void;
   };
 };
 
 type GridEditableState = {
-  isEditing: boolean;
   numColumn: number;
 };
 
 class GridEditable<
   DataRow extends {[key: string]: any},
   ColumnKey extends ObjectKey
-> extends React.Component<GridEditableProps<DataRow, ColumnKey>, GridEditableState> {
+> extends React.Component<GridEditableProps<DataRow, ColumnKey>> {
```

Comment:
`GridEditableState` should be kept because `this.state.numColumn` is still used.